### PR TITLE
FloatingPickers: Move to react-internal

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist

- [x] Addresses an existing issue: Part of #15152
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Move v8 floating picker changes to react-internal, including strict mode fixes from #15131.

### CODE REVIEW

File: packages/react-internal/src/components/FloatingPicker/PeoplePicker/FloatingPeoplePicker.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { SuggestionItemNormal } from './PeoplePickerItems/SuggestionItemDefault'
 import { IPersonaProps } from '../../../Persona';
 import './PeoplePicker.scss';
 import { IBasePickerSuggestionsProps, ISuggestionModel } from '../../../Pickers';
+/* eslint-disable */
```
Comment (ecraig12345) [inline packages/react-internal/src/components/FloatingPicker/PeoplePicker/FloatingPeoplePicker.tsx:8]: You can probably remove this now (react-internal doesn't have the no-explicit-any rule)

File: packages/react-internal/src/components/ExtendedPicker/BaseExtendedPicker.tsx [hunk 2]
```diff
@@ -8,12 +8,12 @@ import { IBaseFloatingPickerProps, BaseFloatingPicker } from '../../FloatingPick
 import { BaseSelectedItemsList, IBaseSelectedItemsListProps } from '../../SelectedItemsList';
 import { FocusZone, FocusZoneDirection } from '../../FocusZone';
 import { Selection, SelectionMode, SelectionZone } from '../../Selection';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (ecraig12345) [inline packages/react-internal/src/components/ExtendedPicker/BaseExtendedPicker.tsx:12]: You can remove the no-explicit-any disables

File: packages/react-internal/src/components/FloatingPicker/Suggestions/Suggestions.types.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import * as React from 'react';
-import { ISuggestionModel } from '../../../Pickers';
+import { ISuggestionModel, ISuggestionItemProps } from '../../../Pickers';
 import { IPersonaProps } from '../../../Persona';
 import { IRefObject } from '../../../Utilities';
-import { ISuggestionItemProps } from '../../pickers/Suggestions/SuggestionsItem.types';
+/* eslint-disable */
```
Comment (ecraig12345) [inline packages/react-internal/src/components/FloatingPicker/Suggestions/Suggestions.types.ts:5]: this too

File: packages/react-internal/src/components/FloatingPicker/Suggestions/SuggestionsControl.tsx [hunk 4]
```diff
@@ -11,6 +11,7 @@ import {
 import { SuggestionsCore } from './SuggestionsCore';
 import * as stylesImport from './SuggestionsControl.scss';
 import { hiddenContentStyle, mergeStyles } from '../../../Styling';
+/* eslint-disable */
```
Comment (ecraig12345) [inline packages/react-internal/src/components/FloatingPicker/Suggestions/SuggestionsControl.tsx:14]: and this

File: packages/react-internal/src/components/FloatingPicker/Suggestions/SuggestionsCore.tsx [hunk 5]
```diff
@@ -3,6 +3,8 @@ import { initializeComponentRef, css } from '../../../Utilities';
 import { ISuggestionItemProps, SuggestionsItem, ISuggestionModel } from '../../../Pickers';
 import { ISuggestionsCoreProps } from './Suggestions.types';
 import * as stylesImport from './SuggestionsCore.scss';
+
+/* eslint-disable */
```
Comment (ecraig12345) [inline packages/react-internal/src/components/FloatingPicker/Suggestions/SuggestionsCore.tsx:7]: this too

**Review by ecraig12345 — APPROVED** [review]
Mostly looks good, thanks! Main feedback is just to remove eslint-disables that aren't needed in this package (especially the file-wide `/* eslint-disable */` since that will prevent it from catching any new errors)

**Review by ecraig12345 — CHANGES_REQUESTED** [review]
Hmm actually I take that back (sorry), the examples need to be updated too. I'm guessing there weren't meaningful changes (since I don't think the public APIs of these components changed much), so you could probably just delete the react-next versions.

**Review by Ridvanovskyy — APPROVED** [review]

**Review by ecraig12345 — APPROVED** [review]

**Pull request merged**